src/core/SamplePlayHandle.cpp

```diff
@@ -76,10 +75,10 @@ SamplePlayHandle::SamplePlayHandle( SampleClip* clip ) :
 
 SamplePlayHandle::~SamplePlayHandle()
 {
-	sharedObject::unref( m_sampleBuffer );
 	if( m_ownAudioPort )
 	{
 		delete audioPort();
+		delete m_sample;
```

Comment:
I don't really like this "conditional ownership" concept which is used in a couple of places in LMMS. I don't mind leaving it here for now, since it was already done for the audio port, but I think we ought to address it in the future.

Reply:
Currently working on refactoring the flow of audio, so this will most likely be addressed then (unless someone wants to address it now; it might be awhile until I complete my work).